src/lib.rs

```diff
@@ -285,22 +285,21 @@ where
 /// let mut rt = RunLoop::new(|| state(|| 0u64));
 ///
 /// let track_wakes = BoolWaker::new();
-/// rt.set_state_change_waker(waker(track_wakes.clone()));
 ///
-/// let (first_commit, first_key) = rt.run_once();
+/// let (first_commit, first_key) = rt.run_once_with(waker(track_wakes.clone()));
```

Comment:
Would it be a problem for `run_once_with` to accept an `impl std::task::Wake` argument and to do this wrapping internally?

Reply:
This is nightly-only feature and would be an overhead if someone would pass an `std::task::Waker`.